### PR TITLE
Update coss api

### PR DESCRIPTION
**Pull request by teekenl**
- What is the purpose of this Pull Request?
- What is the related issue for this Pull Request (if this PR fixes issue, prepend with "Fixes" or "Closes")?
- [ ] I have added Specs
- [ ] (If implementing Market Ticker) I have verified that the `volume` refers to BASE
- [ ] (If implementing Market Ticker) I have verified that the `base` and `target` is assigned correctly
- [ ] I have implemented the `trade_page_url` method that links to the exchange page with the `base` and `target` passed in. If not available, enter the root domain of the exchange website.
- [ ] I have verified at least **ONE** ticker volume matches volume shown on the trading page (use script below)

```
client = Cryptoexchange::Client.new
pairs = client.pairs 'exchange_name'
tickers = pairs.map do |p| client.ticker p end
sorted_tickers = tickers.sort_by do |t| t.volume end.reverse
```

### CODE REVIEW

File: lib/cryptoexchange/exchanges/coss/services/market.rb [hunk 1]
```diff
@@ -17,10 +17,9 @@ def ticker_url
           "#{Cryptoexchange::Exchanges::Coss::Market::API_URL}/ticker"
         end
 
-        def adapt_all(output)
-          result = output['result']
-          result.map do |value|
-            base, target = value['MarketName'].split('-')
+        def adapt_all(result)
+          result.map do |(market_name, value)|
```
Comment (wongy91) [inline lib/cryptoexchange/exchanges/coss/services/market.rb:21]: this is not incorrect, but the codebase for block, there is no bracket for the params.

File: lib/cryptoexchange/exchanges/coss/market.rb [hunk 2]
```diff
@@ -2,10 +2,10 @@ module Cryptoexchange::Exchanges
   module Coss
     class Market < Cryptoexchange::Models::Market
       NAME = 'coss'
-      API_URL = 'https://exchange.coss.io/api'
+      API_URL = 'https://market.coss.io/api'
 
       def self.trade_page_url(args={})
-        "https://coss.io/c/trade?s=#{args[:base].upcase}_#{args[:target].upcase}"
+        "https://market.coss.io/api/trades/#{args[:base].upcase}_#{args[:target].upcase}"
```
Comment (wongy91) [inline lib/cryptoexchange/exchanges/coss/market.rb:8]: this should not gives the api link. It should gives the link where user can operate the trade.

In coss's case, there's no public viewing of trade page unless you are signed-up. Also signed-up is now invitation-only. So, we return `nil`

File: lib/cryptoexchange/exchanges/coss/services/pairs.rb [hunk 3]
```diff
@@ -10,8 +10,8 @@ def fetch
         end
 
         def adapt(output)
-          output['result'].map do |value|
-            base, target = value['MarketName'].split('-')
+          output.map do |(market_name, value)|
```
Comment (wongy91) [inline lib/cryptoexchange/exchanges/coss/services/pairs.rb:13]: same here, no brackets required. `_value` for unused variable.

File: lib/cryptoexchange/exchanges/coss/market.rb (hunk 2 above)
Comment (teekenl) [reply lib/cryptoexchange/exchanges/coss/market.rb:8]: ah i see

File: lib/cryptoexchange/exchanges/coss/services/market.rb (hunk 1 above)
Comment (teekenl) [reply lib/cryptoexchange/exchanges/coss/services/market.rb]: 👍

File: lib/cryptoexchange/exchanges/coss/services/pairs.rb (hunk 3 above)
Comment (teekenl) [reply lib/cryptoexchange/exchanges/coss/services/pairs.rb]: 👍

**Comment by wongy91** [comment]
LGTM 👍

**Pull request merged**